force-app/classes/src/Assertions.cls

```diff
@@ -6,76 +6,142 @@
  */
 @IsTest
 public class Assertions {
-  public virtual class SystemAsserter {
-    public virtual void assertEquals(
-      Object expected,
-      Object actual,
-      String message
-    ) {
-      System.assertEquals(expected, actual, message);
+  public interface MethodSpyAssertions {
+    /**
+     * Assert that a method spy has never been called
+     */
+    void hasNotBeenCalled();
+
+    /**
+     * Assert that a method spy has been called at least one time with any parameters
+     */
+    void hasBeenCalled();
+
+    /**
+     * Assert that a method spy has been called exactly N times with any parameters
+     *
+     * @param count number of times the method spy should have been called
+     */
+    void hasBeenCalledTimes(final Integer count);
+
+    /**
+     * Assert that a method spy has been called at least one time with matching parameters
+     *
+     * @param params params that should match one call paramters
+     * @see          Params
+     */
+    void hasBeenCalledWith(final Params params);
+
+    /**
+     * Assert that a method spy has been lastly called with matching parameters
+     *
+     * @param params params that should match last call paramters
+     * @see          Params
+     */
+    void hasBeenLastCalledWith(final Params params);
+  }
+
+  public interface ErrorMessage {
+    String toString();
+  }
+
+  public interface Asserter {
+    void isTrue(Boolean value, ErrorMessage message);
+    void isFalse(Boolean value, ErrorMessage message);
+  }
+
+  /**
+   * Starts to build assertions on a method spy
+   *
+   * @param spy the MethodSpy to assert
+   * @see       MethodSpy
+   */
+  public static MethodSpyAssertions assertThat(MethodSpy spy) {
+    return assertThat(spy, METHOD_SPY_ASSERTER);
+  }
+
+  /**
+   * Starts to build assertions on a method spy using a custom asserter
+   *
+   * @param spy      the MethodSpy to assert
+   * @param asserter the Assert to use during assertions
+   */
+  @TestVisible
+  private static MethodSpyAssertions assertThat(
+    MethodSpy spy,
+    Asserter asserter
+  ) {
+    return new DefaultMethodSpyAssertions(spy, asserter);
+  }
+
+  @TestVisible
+  private class MethodSpyAsserter implements Asserter {
+    public void isTrue(Boolean value, ErrorMessage message) {
+      if (!value) {
+        Assert.fail(message.toString());
+      }
+    }
+    public void isFalse(Boolean value, ErrorMessage message) {
+      if (value) {
+        Assert.fail(message.toString());
+      }
```

Comment:
question(not blocking): Why not use [Assert.isFalse](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_class_System_Assert.htm#apex_System_Assert_isFalse) ?

Reply:
if we do a Assert.isFalse(value, message.toString()) then the toString() is evaluated even if the message is not used. I did that to avoid time)consuming parsing of the call history in several cases.

Reply:
![image](https://user-images.githubusercontent.com/49166732/204041995-72238b46-ae7d-4a14-a251-e683ec5f351f.png)

Reply:
![giphy](https://user-images.githubusercontent.com/49166732/204042102-1c3d2552-ebc3-4afb-9291-60c710171bd7.gif)